server.py

```diff
@@ -0,0 +1,122 @@
+                    self.broadcast(f"[{datetime.now().strftime(self.DATE_FORM)}]"
+                                   f" {self.usernames_connected[self.clients_connected.index(client)]}:->"
+                                   f" {msg}".encode(self.FORMAT))
+                    logger.info(f"{self.usernames_connected[self.clients_connected.index(client)]}"
+                                f" sent a broadcast message")
+                except:
+                    client.send("Only registered users can post. If you are registered try /login".encode(self.FORMAT))
+
+    def register(self, client, db_path):
+        db = Database(db_path)
+        client.send('Enter your username: '.encode(self.FORMAT))
+        username = client.recv(1024).decode(self.FORMAT)
+        while username in db.get_users_list("users"):
+            client.send("Username is already in use. Try again or /exit".encode(self.FORMAT))
+            username = client.recv(1024).decode(self.FORMAT)
+        client.send('Enter your password: '.encode(self.FORMAT))
+        password = client.recv(1024).decode(self.FORMAT)
+        while len(password) < 6:
+            client.send("Password too short. It should be at lest 5 symbols".encode(self.FORMAT))
+            password = client.recv(1024).decode(self.FORMAT)
+        db.insert_item("users", (username, password))
+        logger.info(f"New user registered. Username is {username}")
+        print(f"New user registered. Username is {username}")
+        client.send(f"You successfully registered as {username}. Now /login.".encode(self.FORMAT))
+
+    def login(self, client, db_path):
+        db = Database(db_path)
+        client.send('Enter your username: '.encode(self.FORMAT))
+        username = client.recv(1024).decode(self.FORMAT)
+        while username not in db.get_users_list("users"):
+            client.send("Wrong username. Try again or use /register or /exit".encode(self.FORMAT))
+            username = client.recv(1024).decode(self.FORMAT)
+        client.send('Enter your password: '.encode(self.FORMAT))
+        password = client.recv(1024).decode(self.FORMAT)
+        while password != db.get_user_password("users", username):
+            client.send("Password is wrong. Try again or use /exit".encode(self.FORMAT))
+            password = client.recv(1024).decode(self.FORMAT)
+        self.usernames_connected.append(username)
+        self.clients_connected.append(client)
+        logger.info(f"{username} logged in")
+        client.send(f"You successfully logged in as {username}.".encode(self.FORMAT))
+
+    def send_to_client(self, msg):
+        addr_username = re.search(r"(?<=@).*?(?=:)", msg).group(0)
+        self.clients_connected[self.usernames_connected.index(addr_username)].send(msg.encode(self.FORMAT))
+
+    def broadcast(self, msg):
+        for client in self.clients_connected:
+            client.send(msg)
+
+
+test_base = Database("users_db")
```

Comment:
why you are creating db instance here and also in server methods?
Create it once and use it

Reply:
Fixed